### PR TITLE
[uwtools_integration] Integrate drivers for make_orog task.

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
## DESCRIPTION OF CHANGES: 

Integrates drivers for make_orog task.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

- [ ] derecho.intel
- [ ] gaea.intel
- [ ] hera.gnu
- [x] hera.intel
- [ ] hercules.intel
- [ ] jet.intel
- [ ] orion.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [x] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## DEPENDENCIES:

Requires a uwtools PR. Coming soon!
## DOCUMENTATION:

n/a

## CHECKLIST

- [ ] My code follows the style guidelines in the Contributor's Guide
- [ ] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by christinaholtNOAA** [comment]
Update: all 6 fundamental tests pass!

File: scripts/make_orog.py
Comment (elcarpenterNOAA) [inline scripts/make_orog.py]: It looks like `sleep` can be removed.

File: ush/config_defaults.yaml
Comment (elcarpenterNOAA) [inline ush/config_defaults.yaml]: This looks like duplication of lines 1438-1441.

Comment (christinaholtNOAA) [reply ush/config_defaults.yaml]: Good eye!

**Review by nbharwani-cu — APPROVED** [review]
LGTM! Great work on this!

File: scripts/make_orog.py
Comment (maddenp-cu) [inline scripts/make_orog.py]: I think could also do this in the `add_argument()` call handling this argument, [like we do in uwtools](https://github.com/ufs-community/uwtools/blob/main/src/uwtools/cli.py#L753).

File: scripts/make_orog.py
Comment (maddenp-cu) [inline scripts/make_orog.py]: Does this work? Isn't `args.key_path` currently still (potentially, at least, per the `--key-path` help string) a dot-delimited string, while drivers expect a [list of strings](https://github.com/ufs-community/uwtools/blob/main/src/uwtools/drivers/driver.py#L50)? But see the suggestion above about converting to a `list[str]` in the `add_argument()`. I mean, I think this will work if the user supplies `--key-path foo`, but will fail if they supply `--key-path foo.bar`.

File: scripts/make_orog.py
Comment (maddenp-cu) [inline scripts/make_orog.py]: Ditto.

File: scripts/make_orog.py
Comment (maddenp-cu) [inline scripts/make_orog.py]: This one uses the split `KEY_PATH`, but it seems like we could dispense with `KEY_PATH` and treat them all `key_path=` kwargs the same.

File: scripts/make_orog.py
Comment (maddenp-cu) [inline scripts/make_orog.py]:
```suggestion
    shave_driver = Shave(
        config=args.config_file,
        key_path=[*args.key_path, sub_path],
```
After doing the split in `add_argument()`.

**Review by WeirAE — APPROVED** [review]
Another solid driver, great work nailing down all the requirements

File: scripts/make_orog.py
Comment (maddenp-cu) [inline scripts/make_orog.py]: Just a thought, but we could
```
>>> from uwtools.api.logging import use_uwtools_logger
>>> use_uwtools_logger()
>>> import logging
>>> logging.info("Hello")
[2024-09-18T14:30:20]     INFO Hello
```
This would get us timestamped output, allow us to also make, when appropriate, `.warning()` and `.error()` calls that might stand out more, and look more like the messaging emitted from `uwtools`. In particular, I like the timestamped output as a user, because I can see how long things take, can compare log output to filesystem timestamps, etc. IMO it can really help when debugging something.

File: scripts/make_orog.py
Comment (maddenp-cu) [inline scripts/make_orog.py:3]: I'm only thinking about this because I was talking to Gillian about it yesterday, but we might consider more use of functions, and at least an `if __name__ == "__main__"` guard, in scripts like this, as otherwise it will be difficult or impossible to generate docs for them, as all the top-level code will run as soon as Sphinx' `autodoc` imports the script as a module to generate docs for it.

File: ush/setup.py
Comment (maddenp-cu) [inline ush/setup.py:227]: I'm obliged to ask: Is `ccpp_suite` really optional here? If not, it would be better to `[ccpp_suite]` and halt here with an error than proceed with an invalid config.

File: parm/wflow/prep.yaml
Comment (maddenp-cu) [inline parm/wflow/prep.yaml]: Mostly curious: Is there a reason to prefer calling `make_orog.py` with `python` explicitly vs adding `#!/usr/bin/env python` to the script, making it executable, and calling it as a program?

File: scripts/make_orog.py
Comment (christinaholtNOAA) [reply scripts/make_orog.py]: I definitely made some bad assumptions here that because in this iteration it's only one level deep, it would only ever be a single string with no dots. I'm making some mods to hopefully fix that poor assumption after your previous comment about splitting while parsing the args.

File: scripts/make_orog.py
Comment (christinaholtNOAA) [reply scripts/make_orog.py]: Yes! Thanks for the suggestion.

File: ush/setup.py
Comment (christinaholtNOAA) [reply ush/setup.py:227]: Yes, I think it could be optional. For example, only 4 of the 8 options are defined in that file. At this point, those are the only ones that need special treatment. It is possible that we will never add options for all the supported suites since one is "default" and may not need additional changes applied anywhere.

File: parm/wflow/prep.yaml
Comment (christinaholtNOAA) [reply parm/wflow/prep.yaml]: I'd be fine either way. I don't personally have a strong opinion on this one, but I would be happy to adopt one if there's an obvious benefit of one over the other that I haven't picked up on yet.

Comment (maddenp-cu) [reply parm/wflow/prep.yaml]: I prefer the `#!` mechanism because invocations are more concise. And I think it's more idiomatic: Although you _can_ run a bash script as `bash a.sh`, this is seldom seen in practice: Usually `a.sh` starts with `#!/usr/bin/env bash` (or similar), the executable bit is set, and the script is invoked as `/path/to/a.sh`.

Better yet, the extension can left off, since the `#!` line defines the interpreter to be used on the file, and editors (emacs and vim at least) honor the `#!` line to set the language type when editing. If these conventions were followed in general then, for example, a Bash script `myscript` could be replaced at any time with a Python script `myscript` and no calling code would have to change. The point of these conventions is to make the implementation details of "programs" invisible: You just run the program by name and it doesn't matter if it's a script in some interpreted language, or a compiled-code executable.

But if any of this is too much for here/now, that's fine.

File: scripts/make_orog.py
Comment (christinaholtNOAA) [reply scripts/make_orog.py:3]: Oh yeah, great point!

File: parm/wflow/prep.yaml
Comment (christinaholtNOAA) [reply parm/wflow/prep.yaml]: I appreciate the well-formed argument for the "why" and am happy to make these changes both here and for the UPP script that already exists.

Comment (maddenp-cu) [reply parm/wflow/prep.yaml]: It could be a demonstration of a helpful practice.

Comment (maddenp-cu) [reply parm/wflow/prep.yaml]: I'm sure you already know, but `git` will track the executable bits as a property of a revision-controlled file, so if you `chmod +x` the file, `git` will see that and allow you to commit that as a change so that the bits will be set when the files are checked out in a clone.

File: scripts/make_orog.py
Comment (christinaholtNOAA) [reply scripts/make_orog.py:3]: I pushed some hefty changes. Let me know what you think.

Comment (maddenp-cu) [reply scripts/make_orog.py:3]: Do you want to keep the `.py` extension? It'll probably never change, but it's not really needed at this point. Your call.

File: scripts/make_orog.py
Comment (maddenp-cu) [inline scripts/make_orog.py:139]: `pylint` may ultimately complain about this, as I think it recommends the
```
logging.error("Error occurred running %s. Please see component error logs.", driver.driver_name())
```
style, with delayed string interpolation. That could be dealt with later if/when it happens.

File: scripts/upp.py
Comment (maddenp-cu) [inline scripts/upp.py:91]: Total nitpick: Some of the `help=` messages end in periods, some don't.

**Review by maddenp-cu — APPROVED** [review]
I added a couple minor comments that you should feel free to ignore. This looks great.

File: scripts/make_orog.py
Comment (christinaholtNOAA) [reply scripts/make_orog.py:139]: We've ignored this error in SRW. 😬

File: scripts/make_orog.py
Comment (christinaholtNOAA) [reply scripts/make_orog.py:3]: I think I will leave it here. It may be a step too far for users, especially NCO. It's pretty standard for systems to use explicit suffixes, even when making the script executable.

File: scripts/make_orog.py
Comment (nbharwani-cu) [inline scripts/make_orog.py:1]: I had a question thinking more about this PR after it merged and when working with `make_sfc_climo`. I was wondering if we could remove the JJOB and exscript for `make_orog` completely now that it runs via uwtools and this python script? Or should we be leaving these files alone?

Comment (christinaholtNOAA) [reply scripts/make_orog.py:1]: Remove them! I totally just forgot. I can open a new PR doing that.

Comment (nbharwani-cu) [reply scripts/make_orog.py:1]: I can remove them as part of my PR if you'd like

File: parm/wflow/prep.yaml
Comment (nbharwani-cu) [inline parm/wflow/prep.yaml:32]: Should I update this line as well in my open PR to `{{ user.MACHINE }}` instead of `hera`?

Comment (christinaholtNOAA) [reply parm/wflow/prep.yaml:32]: Yes please, thank you!

**Pull request merged**
